demo-ling/stock_param_functions.r

```diff
@@ -0,0 +1,47 @@
+## ADDITIONAL FUNCTIONS TO HELP WITH THE SETUP SCRIPTS
+
+## Function to insert stock/species name into g3 parameter
+g3_stock_param <- function(sname, param, lower=NULL, upper=NULL, xby=NULL, sep0="."){
+  
+  nullval <- is.null(lower) + is.null(upper)
+  
+  ## No bounds 
+  if (nullval > 0){
+    if (nullval == 1) warning("One bound is NULL & the other is not NULL, assuming unbounded")
+    tmp <- gadget3:::f_substitute(~g3_param(x), list(x=paste(sname, param, sep=sep0)))
+  }
+  ## Bounds
+  else{
+    lb <- as.numeric(min(lower))
+    ub <- as.numeric(max(upper))
+    tmp <- gadget3:::f_substitute(~bounded(g3_param(x), lb, ub), 
+                                  list(x=paste(sname, param, sep=sep0), lb=lb, ub=ub))
+  }
+  if (!is.null(xby)){
+    tmp <- gadget3:::f_substitute(~y * x, list(y=xby, x=tmp))
+  }
```

Comment:
I'd assume that a ``xby`` parameter behaves like seq in respect to lower and upper, i.e.:

```
> seq(10,20, by = 2)
[1] 10 12 14 16 18 20
```

...whereas this is more like:-

```
> seq(10,20) * 2
 [1] 20 22 24 26 28 30 32 34 36 38 40
```

If this is truly useful (it's only used for setting ``K``), rename it to ``multiplier``?

Reply:
Much better, renamed to `multiplier`